dash/frontend/src/app/core/services/falco.service.ts

```diff
@@ -17,77 +31,77 @@ export class FalcoService {
     private httpClient: HttpClient,
   ) {}
 
-  getFalcoLogs(clusterId: number,
-               limit?: number,
-               page?: number,
-               selectedPriorityLevels?: string [],
-               selectedOrderBy?: string,
-               startDate?: string,
-               endDate?: string,
-               namespace?: string,
-               pod?: string,
-               image?: string
+  getFalcoLogs(clusterId: number, options?: FalcoLogOptions
   ): Observable<IServerResponse<{logCount: number, list: IFalcoLog[]}>> {
-    const params = this.buildParams(clusterId, limit, page, selectedPriorityLevels, selectedOrderBy, startDate, endDate, namespace, pod, image);
+    const params = this.buildParams(clusterId, options);
     return this.httpClient.get('/api/falco', {params});
   }
 
+  getFalcoLogByEventId(eventId: number
+  ): Observable<IServerResponse<IFalcoLog>> {
+    return this.httpClient.get(`/api/falco/${eventId}`);
+  }
+
+  getCountOfFalcoLogsBySignature(clusterId: number, signature: string
+  ): Observable<IServerResponse<IFalcoCount[]>>{
+    const params = this.buildParams(clusterId, {signature} );
+    return this.httpClient.get('/api/falco/count', {params});
+  }
 
   downloadFalcoExport(clusterId: number):
     Observable<IServerResponse<IFalcoCsv>> {
     const params = this.buildParams(clusterId);
     return this.httpClient.get('/api/falco/download', {params});
   }
 
-  buildParams(clusterId: number,
-              limit?: number,
-              page?: number,
-              selectedPriorityLevels?: string [],
-              selectedOrderBy?: string,
-              startDate?: string,
-              endDate?: string,
-              namespace?: string,
-              pod?: string,
-              image?: string
+  buildParams(clusterId: number, options?: FalcoLogOptions
   ): HttpParams {
     let params: HttpParams = new HttpParams();
     params = params.set('cluster-id', clusterId.toString());
 
-    if (limit){
-      params = params.set('limit', String(limit));
+    if (options?.limit){
+      params = params.set('limit', String(options?.limit));
     }
-    if (page){
-      params = params.set('page', String(page));
+    if (options?.page){
+      params = params.set('page', String(options?.page));
     }
 
-    if (selectedPriorityLevels) {
-      selectedPriorityLevels.forEach((priority: string) => {
+    if (options?.selectedPriorityLevels) {
+      options?.selectedPriorityLevels.forEach((priority: string) => {
         params = params.append('priority[]', priority);
       });
     }
 
-    if (selectedOrderBy) {
-      params = params.set('orderBy', selectedOrderBy.toString());
+    if (options?.selectedOrderBy) {
+      params = params.set('orderBy', options?.selectedOrderBy.toString());
```

Comment:
Lets have a list of "allowed" order by options and not allow them to select anything else. In SQL, the order by is a field name so it isn't escaped the same way. Maybe the framework handles this for us, or maybe not, but I find that making sure its a list of allowed ordered by options is the most foolproof way to prevent sql injection.

Reply:
I have the list of order by options here https://github.com/m9sweeper/m9sweeper/blob/main/dash/frontend/src/app/modules/private/pages/falco/falco-events-list/falco-events-list.component.ts. Is it what you looking for?

Reply:
No, we need to also validate it in the backend because a nefarious user could pass anything to the backend (regardless of what list we show in the frontend).